modules/new_serverpod_auth/serverpod_auth_email_account/serverpod_auth_email_account_server/lib/src/business/password_hash.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:pointycastle/key_derivators/api.dart';
+import 'package:pointycastle/key_derivators/argon2.dart';
+import 'package:serverpod_auth_email_account_server/src/business/email_account_secrets.dart';
+
+/// Class for handling password hashing.
+///
+/// Uses the Argon2id algorithm.
+/// See: https://en.wikipedia.org/wiki/Argon2
+abstract final class PasswordHash {
+  /// Prefix for Argon2 Hash with lower-case email as the `salt`, and a configured `pepper`
+  static const _prefix = 'argon2:';
+
+  /// Create the password hash for the given [email] / [password] pair.
+  static String createHash({
+    required final String email,
+    required final String password,
+  }) {
+    if (email != email.toLowerCase()) {
+      throw ArgumentError.value(
+        email,
+        'email',
+        'Email is not given in lower-case.',
+      );
+    }
+
+    final pepper = EmailSecrets.pepper ?? ''; // TODO: Forbid `null`?
```

Comment:
remove TODO, but we should really forbid this 😅

Reply:
Not providing the secrets now fails in non-`test` run modes. (I would also have it fail in tests, but did not see a way to provide the password to `withServerpod`. Thus leaving it open, as maybe we can support it.)